scripts/checkpatch.sh

```diff
@@ -50,7 +50,23 @@ case "$op" in
 		;;
 	*)
 		echo "Checking commit(s):"
-		for c in $*; do checkpatch $c; done
+    version="$(git --version | awk -F '.' '{print $2}')"
+    if (( version < 19 )); then
+      for c in "$@"; do checkpatch "$c"; done
+    else
+      for arg in "$@"; do
+        # parse the argument into an object
+        object="$(git rev-parse "${arg}")" || continue
```

Comment:
Wouldn't `git log --format=%H ${arg}` work just as well, but without the special case below?
Or is the `git rev-parse` syntax more powerful?

Reply:
When given a single commit id or any other object, both `git log` and `git rev-list` will print the object's entire git history in reverse chronological order. I used `git rev-parse` to check if the argument is valid and to parse the input into a single commit id or an output understood by `git rev-list` which converts it into a list of commits. Then, I use `git cat-file` to check if the output needs to be converted by `git rev-list` or if I can send the id straight to checkpatch.